### PR TITLE
Test/event creation e2e

### PR DESCRIPTION
**Pull request by Zafouche**
## Description

This PR implements an end-to-end test for the `Event` creation mechanism in our application. The test should automate the entire process of creating a new Event, including filling out all required fields, handling the image picker (if I am successful in finding how to do this, which should be possible thanks to `Espresso`'s `Intents`), selecting dates and times, and keeping a completely mocked environment thanks to `okhttp3`'s mocked web server.

## Motivation and Context

Adding an e2e test for the Event creation process ensures that the user story remains respected. It helps testing our feature in the actual app's context.

### EventCreationE2ETest:

- Form Filling: Inputs test data into all required fields of the Event creation form.
- Image Picker Handling: Mocks the image picker result to avoid external dependencies and selects a test image that exists in our `res/drawable` folder. 
- Date and Time Selection: Selects specific dates and times using the DatePicker and TimePicker components. 
- Location Search Integration: Mocks the Nominatim API responses using MockWebServer to simulate location search and selection. 
- Event Verification: Verifies that the Event is successfully created and appears on the Home screen. 

### MockWebServer Integration:

- Setup for Nominatim API: Configured MockWebServer to provide mock responses for the Nominatim API during tests.

### Hilt Dependency Injection:

- TestNetworkModule: Created a TestNetworkModule that provides test-specific dependencies, e.g. the MockWebServer and the mocked NominatimApiService.

### CODE REVIEW

**Comment by Zafouche** [comment]
The test now entirely passes locally, thanks to these changes:
- setDateTime now works. It uses Material3's node contents to set a date and a time. After doing a lot of research, it turns out `Espresso` or `uiAutomator` only work with `Android`'s DatePicker and TimePicker. The [only existing solution](https://stackoverflow.com/questions/77111152/jetpack-compose-ui-test-with-material3-datepickerdialog) that I found is to print out all the available nodes when `Material3`'s dialog pops up to see how the elements that we want to click on are defined. Thanks to that, I figured out how to automate the pickers.
- The image picker intent has also been adapted to get the correct URI.
- The MockWebServer has been adapted to use a different port than the one used by Firebase emulator.
- Added logic that checks that the event has been correctly by verifying it appears on the Home screen.

**Comment by Zafouche** [comment]
`connectedCheck` passes locally for me, but not on the CI :/

**Comment by Zafouche** [comment]
🎉

**Review by oskar-codes — APPROVED** [review]
This looks perfectly good to me, not much to say.

File: app/src/androidTest/java/com/android/unio/HiltTestUtility.kt [hunk 1]
```diff
@@ -13,6 +15,11 @@ import dagger.hilt.android.testing.HiltTestApplication
  * configure the test application for Hilt.
  */
 class HiltApplication : AndroidJUnitRunner() {
+  override fun onCreate(arguments: Bundle) {
+    StrictMode.setThreadPolicy(StrictMode.ThreadPolicy.Builder().permitAll().build())
+    super.onCreate(arguments)
+  }
```
[the remark this answers is not in the document]
Comment (Zafouche) [reply app/src/androidTest/java/com/android/unio/HiltTestUtility.kt:21]: It allows us to perform network operations on the main thread. By default, the `AndroidJUnitRunner` does not allow it. Considering it is only utilized in this e2e test, it's fine to do so.

**Pull request merged**